pkgs/development/python-modules/miasm/default.nix

```
@@ -0,0 +1,51 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, lib
+, future
+, gcc
+, llvmlite
+, parameterized
+, pycparser
+, pyparsing
+, z3-solver
+}:
+buildPythonPackage rec {
+  pname = "miasm";
+  version = "0.1.3-unstable-2024-04-28";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "cea-sec";
+    repo = "miasm";
+    rev = "90dc1671b59077ee27c3d44d9d536d6659eb3bbe";
+    hash = "sha256-nkRcirJLmTwSL7lwd+Yk6mteU3YPnm5ekJ4eFF5FmYo=";
+  };
+
+  patches = [
+    #  Use a valid semver as now required by setuptools
+    ./0001-setup.py-use-valid-semver.patch
+
+    # Removes the (unfree) IDAPython dependency
+    ./0002-core-remove-IDAPython-dependency.patch
+  ];
+
+  propagatedBuildInputs = [
+    future
+    llvmlite
+    parameterized
+    pycparser
+    pyparsing
+    z3-solver
+  ];
+
+  dependencies = [ gcc ];
+
+  pythonImportsCheck = [ "miasm" ];
+
+  meta = {
+    description = "Reverse engineering framework in Python ";
+    homepage = "https://github.com/cea-sec/miasm";
+    license = lib.licenses.gpl2;
```

According to a couple of file headers (e.g., https://github.com/cea-sec/miasm/blob/master/miasm/os_dep/win_api_x86_32_seh.py), the license is GPLv2+.

```suggestion
    license = lib.licenses.gpl2Plus;
```